src/main/java/org/jboss/pnc/konfluxbuilddriver/Driver.java

```diff
@@ -86,6 +107,18 @@ public BuildResponse create(BuildRequest buildRequest) {
         // TODO: This should be changed to true eventually.
         templateProperties.put("ENABLE_INDY_PROXY", config.indyProxyEnabled());
 
+        try {
+            Request notificationCallback = new Request(
+                    Request.Method.PUT,
+                    new URI(StringUtils.appendIfMissing(config.selfBaseUrl(), "/") + "internal/completed"),
```

Comment:
It seems the "internal/completed" REST api calls the Driver's "completed()" method which calls the "internal/completed" endpoint again... would this be an infinite loop?

Reply:
Sorry I missed this. The `build` endpoint takes a callback (to call back to PNC). That callback is embedded into the pipeline which is then passed to the `PipelineNotification` so the driver completed method should be instead calling the very first callback, not the internal endpoint.